<a href="https://colab.research.google.com/github/mamartinezg/mamartinezg/blob/main/Copia_de_Copia_de_Untitled1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import numpy as np
import pandas as pd
from sklearn.ensemble import RandomForestClassifier
import matplotlib.pyplot as plt
import seaborn as sns

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


A continuación procedemos a leer las dos bases de datos la que relaciona historial academico del estudiante y la que tiene los resultados de valor agregado de los estudiantes.

In [ ]:
ruta = pd.ExcelFile('/ValorAgregado_Datos completos_data.xlsx')
VA = pd.read_excel(ruta, 'ValorAgregado_Datos completos_d')

<ipython-input-3-85543ff33a41>:2: FutureWarning: Inferring datetime64[ns] from data containing strings is deprecated and will be removed in a future version. To retain the old behavior explicitly pass Series(data, dtype=datetime64[ns])
  VA = pd.read_excel(ruta, 'ValorAgregado_Datos completos_d')


In [ ]:
pd.set_option('display.max_columns', None)
VA.head()

Anosbpro Categoria_val_agreg  \
0      2018               Va-In   
1      2018               Va-Lc   
2      2018               Va-Rc   
3      2018               Va-In   
4      2018               Va-Lc   

                                 Estu Prgm Academico  valor_agreg_categoria  \
0  LICENCIATURA EN EDUCACIÓN BÁSICA CON ÉNFASIS E...             -44.351329   
1  LICENCIATURA EN EDUCACIÓN BÁSICA CON ÉNFASIS E...             -48.781156   
2  LICENCIATURA EN EDUCACIÓN BÁSICA CON ÉNFASIS E...             -36.705174   
3  ADMINISTRACIÓN EN FINANZAS Y NEGOCIOS INTERNAC...               7.242769   
4  ADMINISTRACIÓN EN FINANZAS Y NEGOCIOS INTERNAC...              -2.997487   

   Anosb11  Añosb112 Cole Caracter Cole Naturaleza     Edadsbpro        Epcc  \
0     2013     20132     ACADÉMICO         OFICIAL  21 o 22 Años  133.124611   
1     2013     20132     ACADÉMICO         OFICIAL  21 o 22 Años  133.124611   
2     2013     20132     ACADÉMICO         OFICIAL  21 o 22 Años  133.124611   
3     2013     20132     ACADÉMICO         OFICIAL  21 o 22 Años  150.570796   
4     2013     20132     ACADÉMICO         OFICIAL  21 o 22 Años  150.570796   

         Epce        Epin        Eplc        Eprc     Estu Areareside  \
0  144.908047  123.351329  144.781156  140.705174          Area Rural   
1  144.908047  123.351329  144.781156  140.705174          Area Rural   
2  144.908047  123.351329  144.781156  140.705174          Area Rural   
3  148.259827  141.757231  152.997487  157.645090  Cabecera Municipal   
4  148.259827  141.757231  152.997487  157.645090  Cabecera Municipal   

  Estu Consecutivo Estu Consecutivo 11 Estu Depto Reside Estu Estadocivil  \
0   EK201830168341    SB11201320062403           CORDOBA          Soltero   
1   EK201830168341    SB11201320062403           CORDOBA          Soltero   
2   EK201830168341    SB11201320062403           CORDOBA          Soltero   
3   EK201830178478    SB11201320216423           CORDOBA          Soltero   
4   EK201830178478    SB11201320216423           CORDOBA          Soltero   

  Estu Fechanacimiento Estu Genero Estu Horassemanatrabaja Estu Mcpio Reside  \
0           1997-03-09           M         Más de 30 horas        SAN PELAYO   
1           1997-03-09           M         Más de 30 horas        SAN PELAYO   
2           1997-03-09           M         Más de 30 horas        SAN PELAYO   
3           1997-10-31           M     Entre 11 y 20 horas      PLANETA RICA   
4           1997-10-31           M     Entre 11 y 20 horas      PLANETA RICA   

   Estu Nse Ies  Estu Nse Individual Estu Nucleo Pregrado  \
0             1                  1.0            EDUCACIÓN   
1             1                  1.0            EDUCACIÓN   
2             1                  1.0            EDUCACIÓN   
3             1                  2.0       ADMINISTRACIÓN   
4             1                  2.0       ADMINISTRACIÓN   

  Estu Pagomatriculabeca Estu Pagomatriculacredito Estu Pagomatriculapadres  \
0                     Si                        No                       Si   
1                     Si                        No                       Si   
2                     Si                        No                       Si   
3                     Si                        Si                       Si   
4                     Si                        Si                       Si   

  Estu Pagomatriculapropio  Fami Cuartoshogar  \
0                       No                4.0   
1                       No                4.0   
2                       No                4.0   
3                       No                3.0   
4                       No                3.0   

                  Fami Educacionmadre                 Fami Educacionpadre  \
0                   Primaria completa                   Primaria completa   
1                   Primaria completa                   Primaria completa   
2                   Primaria completa                   Primaria completa   
3  Secun

In [ ]:
xls = pd.ExcelFile('/Data_Estudiantes_2015-1_2021-1.xlsx')
df = pd.read_excel(xls, 'Data_2015-1_2021-1')

In [ ]:
df.head()

Id_Usuario Fuente_Datos Año_Periodo Session  \
0  P000136783     INSCRITO      2020-2    PREG   
1  P000125627     INSCRITO      2020-2    PREG   
2  P000136783     INSCRITO      2020-1    PREG   
3  P000130749     INSCRITO      2020-1    PREG   
4  P000125627     INSCRITO      2020-1    PREG   

                                   Facultad  \
0  FACULTAD DE EDUCACIÓN Y CIENCIAS HUMANAS   
1  FACULTAD DE EDUCACIÓN Y CIENCIAS HUMANAS   
2  FACULTAD DE EDUCACIÓN Y CIENCIAS HUMANAS   
3          FACULTAD DE CIENCIAS DE LA SALUD   
4  FACULTAD DE EDUCACIÓN Y CIENCIAS HUMANAS   

                   Departamento_Programa              Nombre_Programa  \
0  DEPARTAMENTO DE INFORMÁTICA EDUCATIVA  LICENCIATURA EN INFORMÁTICA   
1  DEPARTAMENTO DE INFORMÁTICA EDUCATIVA  LICENCIATURA EN INFORMÁTICA   
2  DEPARTAMENTO DE INFORMÁTICA EDUCATIVA  LICENCIATURA EN INFORMÁTICA   
3          DEPARTAMENTO DE BACTERIOLOGÍA                BACTERIOLOGÍA   
4  DEPARTAMENTO DE INFORMÁTICA EDUCATIVA  LICENCIATURA EN INFORMÁTICA   

       Sede Categoria Circunscripcion     Genero Estado_Civil Factor_RH  \
0  MONTERÍA  ADMITIDO         REGULAR   FEMENINO   Soltero(a)        O+   
1  MONTERÍA  ADMITIDO         REGULAR  MASCULINO   Soltero(a)        B+   
2  MONTERÍA  ADMITIDO         REGULAR   FEMENINO   Soltero(a)        O+   
3  MONTERÍA  ADMITIDO         REGULAR  MASCULINO   Soltero(a)        O+   
4  MONTERÍA  ADMITIDO         REGULAR  MASCULINO   Soltero(a)        B+   

      Fecha_Nacimiento Estrato      Pais Departamento      Ciudad    Estatura  \
0  2002-06-14 00:00:00       1  COLOMBIA      CÓRDOBA  SAN PELAYO         167   
1  2000-07-13 00:00:00       1  COLOMBIA      CÓRDOBA      AYAPEL         180   
2  2002-06-14 00:00:00       1  COLOMBIA      CÓRDOBA  SAN PELAYO         167   
3  2002-02-18 00:00:00       1  COLOMBIA      CÓRDOBA  BUENAVISTA  NO REPORTA   
4  2000-07-13 00:00:00       1  COLOMBIA      CÓRDOBA      AYAPEL         180   

  Sisben Descripcion_Sisben Caracter    Enfasis          No SNP Puntaje  \
0      3       NO REPORTADO  PUBLICA  ACADÉMICO  AC201943280203     225   
1      1       NO REPORTADO  PUBLICA  ACADÉMICO  AC201722088367  277.25   
2      3       NO REPORTADO  PUBLICA  ACADÉMICO  AC201943280203     225   
3      1       NO REPORTADO  PUBLICA  ACADÉMICO  AC201945270699   242.5   
4      1       NO REPORTADO  PUBLICA  ACADÉMICO  AC201722088367  277.25   

  Ubicacion_Semestral Promedio_Acumulado Promedio_Semestre  Periodo_Admision  \
0                 NaN         NO REPORTA         NO REPORTA      2021 - SEM1   
1                 NaN         NO REPORTA         NO REPORTA      2021 - SEM1   
2                 NaN         NO REPORTA         NO REPORTA      2021 - SEM1   
3                 NaN         NO REPORTA         NO REPORTA      2021 - SEM1   
4                 NaN         NO REPORTA         NO REPORTA      2021 - SEM1   

  Periodos_Cursados                 Modalidad Acceso_Internet Portatil_PC  \
0                 0  UNIVERSITARIO PRESENCIAL              NO          NO   
1                 0  UNIVERSITARIO PRESENCIAL              NO          NO   
2                 0  UNIVERSITARIO PRESENCIAL              NO          NO   
3                 0  UNIVERSITARIO PRESENCIAL              SI          SI   
4                 0  UNIVERSITARIO PRESENCIAL              NO          NO   

  Celular_Tablet     Grupo_Etnico Persona_Con_Discapacidad  \
0             SI     NO PERTENECE                       NO   
1             SI     NO PERTENECE                       NO   
2             SI     NO PERTENECE                       NO   
3             SI  PUEBLO INDÍGENA                       NO   
4             SI     NO PERTENECE                       NO   

  Capacidad_Excepcional                         Poblacion_Vulnerable  \
0            NO REPORTA                                   NO REPORTA   
1            NO REPORTA                                   NO REPORTA   
2            NO REPORTA                                   


En el siguiente código se filtra los registros que tienen Fuente_Datos igual a 'MATRICULADO', y luego realiza el proceso de eliminación de duplicados para mantener el último Año_Periodo en esos registros.

In [ ]:
df = df[df['Fuente_Datos'] == 'MATRICULADO']
df['Año_Periodo'] = pd.to_datetime(df['Año_Periodo'])
df.sort_values(by=['Id_Usuario', 'Año_Periodo'], ascending=[True, False], inplace=True)
df.drop_duplicates(subset='Id_Usuario', keep='first', inplace=True)
df.reset_index(drop=True, inplace=True)

Ahora se procede a realizar el cruce de las dos bases de datos para esto se toma como llave programa academico y año, pues estos registros estan en ambas bases de datos.

In [ ]:
# Realiza una combinación basada en las columnas llave es decir programa academico y año
VA['Estu Fechanacimiento'] = pd.to_datetime(VA['Estu Fechanacimiento'])
df['Fecha_Nacimiento'] = pd.to_datetime(df['Fecha_Nacimiento'],errors='coerce')
df = df.drop_duplicates(subset=['Fecha_Nacimiento', 'Nombre_Programa'])
DataVA = VA.merge(df, left_on=['Estu Fechanacimiento', 'Estu Prgm Academico'],
                  right_on=['Fecha_Nacimiento', 'Nombre_Programa'], how='left')

<ipython-input-8-7b95498152da>:3: UserWarning: Parsing dates in DD/MM/YYYY format when dayfirst=False (the default) was specified. This may lead to inconsistently parsed dates! Specify a format to ensure consistent parsing.
  df['Fecha_Nacimiento'] = pd.to_datetime(df['Fecha_Nacimiento'],errors='coerce')


In [ ]:
DataVA['Promedio_Acumulado'].isna().sum()

1059

In [ ]:
DataVA['Edadsbpro'].isna().sum()

4473

In [ ]:
DataVA['Departamento'].isna().sum()

1050

In [ ]:
DataVA['Estu Pagomatriculabeca'].isna().sum()

4473

In [ ]:
DataVA.head()

Anosbpro Categoria_val_agreg  \
0      2018               Va-In   
1      2018               Va-Lc   
2      2018               Va-Rc   
3      2018               Va-In   
4      2018               Va-Lc   

                                 Estu Prgm Academico  valor_agreg_categoria  \
0  LICENCIATURA EN EDUCACIÓN BÁSICA CON ÉNFASIS E...             -44.351329   
1  LICENCIATURA EN EDUCACIÓN BÁSICA CON ÉNFASIS E...             -48.781156   
2  LICENCIATURA EN EDUCACIÓN BÁSICA CON ÉNFASIS E...             -36.705174   
3  ADMINISTRACIÓN EN FINANZAS Y NEGOCIOS INTERNAC...               7.242769   
4  ADMINISTRACIÓN EN FINANZAS Y NEGOCIOS INTERNAC...              -2.997487   

   Anosb11  Añosb112 Cole Caracter Cole Naturaleza     Edadsbpro        Epcc  \
0     2013     20132     ACADÉMICO         OFICIAL  21 o 22 Años  133.124611   
1     2013     20132     ACADÉMICO         OFICIAL  21 o 22 Años  133.124611   
2     2013     20132     ACADÉMICO         OFICIAL  21 o 22 Años  133.124611   
3     2013     20132     ACADÉMICO         OFICIAL  21 o 22 Años  150.570796   
4     2013     20132     ACADÉMICO         OFICIAL  21 o 22 Años  150.570796   

         Epce        Epin        Eplc        Eprc     Estu Areareside  \
0  144.908047  123.351329  144.781156  140.705174          Area Rural   
1  144.908047  123.351329  144.781156  140.705174          Area Rural   
2  144.908047  123.351329  144.781156  140.705174          Area Rural   
3  148.259827  141.757231  152.997487  157.645090  Cabecera Municipal   
4  148.259827  141.757231  152.997487  157.645090  Cabecera Municipal   

  Estu Consecutivo Estu Consecutivo 11 Estu Depto Reside Estu Estadocivil  \
0   EK201830168341    SB11201320062403           CORDOBA          Soltero   
1   EK201830168341    SB11201320062403           CORDOBA          Soltero   
2   EK201830168341    SB11201320062403           CORDOBA          Soltero   
3   EK201830178478    SB11201320216423           CORDOBA          Soltero   
4   EK201830178478    SB11201320216423           CORDOBA          Soltero   

  Estu Fechanacimiento Estu Genero Estu Horassemanatrabaja Estu Mcpio Reside  \
0           1997-03-09           M         Más de 30 horas        SAN PELAYO   
1           1997-03-09           M         Más de 30 horas        SAN PELAYO   
2           1997-03-09           M         Más de 30 horas        SAN PELAYO   
3           1997-10-31           M     Entre 11 y 20 horas      PLANETA RICA   
4           1997-10-31           M     Entre 11 y 20 horas      PLANETA RICA   

   Estu Nse Ies  Estu Nse Individual Estu Nucleo Pregrado  \
0             1                  1.0            EDUCACIÓN   
1             1                  1.0            EDUCACIÓN   
2             1                  1.0            EDUCACIÓN   
3             1                  2.0       ADMINISTRACIÓN   
4             1                  2.0       ADMINISTRACIÓN   

  Estu Pagomatriculabeca Estu Pagomatriculacredito Estu Pagomatriculapadres  \
0                     Si                        No                       Si   
1                     Si                        No                       Si   
2                     Si                        No                       Si   
3                     Si                        Si                       Si   
4                     Si                        Si                       Si   

  Estu Pagomatriculapropio  Fami Cuartoshogar  \
0                       No                4.0   
1                       No                4.0   
2                       No                4.0   
3                       No                3.0   
4                       No                3.0   

                  Fami Educacionmadre                 Fami Educacionpadre  \
0                   Primaria completa                   Primaria completa   
1                   Primaria completa                   Primaria completa   
2                   Primaria completa                   Primaria completa   
3  Secun

In [ ]:
# Seleccionamos las columnas de interés
columnas_de_interes = ['Anosbpro', 'Categoria_val_agreg','Estu Prgm Academico',	'valor_agreg_categoria', 'Cole Naturaleza',	'Edadsbpro', 'Estu Areareside',
                       'Estu Estadocivil', 'Estu Genero', 'Estu Pagomatriculabeca',	'Fami Educacionmadre', 'Fami Educacionpadre',	'Fami Estratovivienda',	'Fami Ingresofmiliarmensual',
                       'Fami Tienecomputador', 'Fami Tieneinternet',	'Indice_crecimiento',	'Inst Nombre Institucion',	'Mod Competen Ciudada Punt',	'Mod Comuni Escrita Punt',
                       'Mod Ingles Punt',	'Mod Lectura Critica Punt',	'Mod Razona Cuantitat Punt',	'Punt C Naturales',	'Punt Ingles',	'Punt Lectura Critica',	'Punt Matematicas',
                       'Punt Sociales Ciudadanas', 'Puntaje,Global,SB11', 'Sede', 'Categoria',	'Estado_Civil',	'Estrato', 'Departamento',	'Ciudad',	'Caracter',	'No SNP',
                       'Puntaje',	'Ubicacion_Semestral',	'Promedio_Acumulado',	'Modalidad',	'Acceso_Internet',	'Portatil_PC',	'Celular_Tablet',	'Grupo_Etnico',
                       'Persona_Con_Discapacidad',	'Capacidad_Excepcional',	'Poblacion_Vulnerable',	'Inmigrantes',	'Tipo_Discapacidad', 'Tiene_Hijos',	'Consume_Sustancias_Psicoactivas']

# Nuevo DataFrame con las columnas de interés
DataVA_Final= DataVA[columnas_de_interes]


In [ ]:
DataVA_Final.head()

Anosbpro Categoria_val_agreg  \
0      2018               Va-In   
1      2018               Va-Lc   
2      2018               Va-Rc   
3      2018               Va-In   
4      2018               Va-Lc   

                                 Estu Prgm Academico  valor_agreg_categoria  \
0  LICENCIATURA EN EDUCACIÓN BÁSICA CON ÉNFASIS E...             -44.351329   
1  LICENCIATURA EN EDUCACIÓN BÁSICA CON ÉNFASIS E...             -48.781156   
2  LICENCIATURA EN EDUCACIÓN BÁSICA CON ÉNFASIS E...             -36.705174   
3  ADMINISTRACIÓN EN FINANZAS Y NEGOCIOS INTERNAC...               7.242769   
4  ADMINISTRACIÓN EN FINANZAS Y NEGOCIOS INTERNAC...              -2.997487   

  Cole Naturaleza     Edadsbpro     Estu Areareside Estu Estadocivil  \
0         OFICIAL  21 o 22 Años          Area Rural          Soltero   
1         OFICIAL  21 o 22 Años          Area Rural          Soltero   
2         OFICIAL  21 o 22 Años          Area Rural          Soltero   
3         OFICIAL  21 o 22 Años  Cabecera Municipal          Soltero   
4         OFICIAL  21 o 22 Años  Cabecera Municipal          Soltero   

  Estu Genero Estu Pagomatriculabeca                 Fami Educacionmadre  \
0           M                     Si                   Primaria completa   
1           M                     Si                   Primaria completa   
2           M                     Si                   Primaria completa   
3           M                     Si  Secundaria (Bachillerato) completa   
4           M                     Si  Secundaria (Bachillerato) completa   

                  Fami Educacionpadre Fami Estratovivienda  \
0                   Primaria completa            Estrato 1   
1                   Primaria completa            Estrato 1   
2                   Primaria completa            Estrato 1   
3  Secundaria (Bachillerato) completa            Estrato 2   
4  Secundaria (Bachillerato) completa            Estrato 2   

  Fami Ingresofmiliarmensual Fami Tienecomputador Fami Tieneinternet  \
0            Menos de 1 SMLV                   No                 No   
1            Menos de 1 SMLV                   No                 No   
2            Menos de 1 SMLV                   No                 No   
3            Menos de 1 SMLV                   Si                 Si   
4            Menos de 1 SMLV                   Si                 Si   

   Indice_crecimiento          Inst Nombre Institucion  \
0            0.300000  UNIVERSIDAD DE CORDOBA-MONTERIA   
1            0.300000  UNIVERSIDAD DE CORDOBA-MONTERIA   
2            0.350000  UNIVERSIDAD DE CORDOBA-MONTERIA   
3            0.548872  UNIVERSIDAD DE CORDOBA-MONTERIA   
4            0.518797  UNIVERSIDAD DE CORDOBA-MONTERIA   

   Mod Competen Ciudada Punt  Mod Comuni Escrita Punt  Mod Ingles Punt  \
0                       99.0                     57.0             79.0   
1                       99.0                     57.0             79.0   
2                       99.0                     57.0             79.0   
3                      156.0                     65.0            149.0   
4                      156.0                     65.0            149.0   

   Mod Lectura Critica Punt  Mod Razona Cuantitat Punt  Punt C Naturales  \
0                      96.0                        104                58   
1                      96.0                        104                58   
2                      96.0                        104                58   
3                     150.0                        159                65   
4                     150.0                        159                65   

   Punt Ingles  Punt Lectura Critica  Punt Matematicas  \
0           40                    56                50   
1           40                    56                50   
2           40                    56                50   
3           44                    58                53   
4           44                    58                53   

   Punt Sociales Ciud

In [ ]:
filas, columnas = DataVA_Final.shape
print("Número de filas:", filas)
print("Número de columnas:", columnas)

Número de filas: 19680
Número de columnas: 52


In [ ]:
DataVA_Final.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 19680 entries, 0 to 19679
Data columns (total 52 columns):
 #   Column                           Non-Null Count  Dtype  
---  ------                           --------------  -----  
 0   Anosbpro                         19680 non-null  int64  
 1   Categoria_val_agreg              19680 non-null  object 
 2   Estu Prgm Academico              19680 non-null  object 
 3   valor_agreg_categoria            19680 non-null  float64
 4   Cole Naturaleza                  19680 non-null  object 
 5   Edadsbpro                        15207 non-null  object 
 6   Estu Areareside                  19680 non-null  object 
 7   Estu Estadocivil                 4779 non-null   object 
 8   Estu Genero                      19680 non-null  object 
 9   Estu Pagomatriculabeca           15207 non-null  object 
 10  Fami Educacionmadre              15207 non-null  object 
 11  Fami Educacionpadre              15207 non-null  object 
 12  Fami Estratovivien

In [ ]:
data_types = DataVA.dtypes
data_types

Anosbpro                             int64
Categoria_val_agreg                 object
Estu Prgm Academico                 object
valor_agreg_categoria              float64
Anosb11                              int64
                                    ...   
Tipo_Discapacidad                   object
Estado_de_Embarazo                  object
Tiene_Hijos                         object
Soltera_Cabeza_Familia              object
Consume_Sustancias_Psicoactivas     object
Length: 102, dtype: object

Veamos si hay valores nulos en nuestra data

In [ ]:
columnas_con_na = DataVA_Final.columns[DataVA_Final.isna().any()].tolist()
columnas_con_na

['Edadsbpro',
 'Estu Estadocivil',
 'Estu Pagomatriculabeca',
 'Fami Educacionmadre',
 'Fami Educacionpadre',
 'Fami Estratovivienda',
 'Fami Ingresofmiliarmensual',
 'Fami Tienecomputador',
 'Fami Tieneinternet',
 'Mod Competen Ciudada Punt',
 'Mod Comuni Escrita Punt',
 'Mod Ingles Punt',
 'Mod Lectura Critica Punt',
 'Puntaje,Global,SB11',
 'Sede',
 'Categoria',
 'Estado_Civil',
 'Estrato',
 'Departamento',
 'Ciudad',
 'Caracter',
 'No SNP',
 'Puntaje',
 'Ubicacion_Semestral',
 'Promedio_Acumulado',
 'Modalidad',
 'Acceso_Internet',
 'Portatil_PC',
 'Celular_Tablet',
 'Grupo_Etnico',
 'Persona_Con_Discapacidad',
 'Capacidad_Excepcional',
 'Poblacion_Vulnerable',
 'Inmigrantes',
 'Tipo_Discapacidad',
 'Tiene_Hijos',
 'Consume_Sustancias_Psicoactivas']

In [ ]:
# Verificar si hay valores perdidos en el DataFrame
hay_valores_perdidos = DataVA_Final.isna().any().any()

# Mostrar el resultado
if hay_valores_perdidos:
    print("El DataFrame contiene valores perdidos.")
else:
    print("El DataFrame no contiene valores perdidos.")

El DataFrame contiene valores perdidos.


In [ ]:
# Verificar si la columna tiene valores perdidos
hay_valores_perdidos = DataVA_Final['Promedio_Acumulado'].isna().any()

# Mostrar el resultado
if hay_valores_perdidos:
    print(f"La columna Promedio_Acumulado tiene valores perdidos.")
else:
    print(f"La columna Promedio_Acumulado no tiene valores perdidos.")

La columna Promedio_Acumulado tiene valores perdidos.


In [ ]:
DataVA_Final['Promedio_Acumulado'] = pd.to_numeric(DataVA_Final['Promedio_Acumulado'], errors='coerce')


<ipython-input-20-b94d38f4053e>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  DataVA_Final['Promedio_Acumulado'] = pd.to_numeric(DataVA_Final['Promedio_Acumulado'], errors='coerce')


In [ ]:
# Primero, calcula los promedios por la combinación única de 'Anosbpro' y 'Estu Prgm Academico' solo para las filas con valores no perdidos en 'Promedio_Acumulado'
promedios = DataVA_Final.dropna(subset=['Promedio_Acumulado']).groupby(['Anosbpro', 'Estu Prgm Academico'])['Promedio_Acumulado'].mean().reset_index()
promedios.head()

Anosbpro                                Estu Prgm Academico  \
0      2018                                        ACUICULTURA   
1      2018  ADMINISTRACIÓN EN FINANZAS Y NEGOCIOS INTERNAC...   
2      2018                            ADMINISTRACIÓN EN SALUD   
3      2018                                      BACTERIOLOGÍA   
4      2018                                           BIOLOGÍA   

   Promedio_Acumulado  
0            3.612000  
1            3.626357  
2            3.708190  
3            3.756000  
4            3.550000

In [ ]:
# Luego, crea un diccionario con estos valores promedio
promedio_imputacion = dict(zip(zip(promedios['Anosbpro'], promedios['Estu Prgm Academico']), promedios['Promedio_Acumulado']))
promedio_imputacion

{(2018, 'ACUICULTURA'): 3.6119999999999997,
 (2018,
  'ADMINISTRACIÓN EN FINANZAS Y NEGOCIOS INTERNACIONALES'): 3.626356589147287,
 (2018, 'ADMINISTRACIÓN EN SALUD'): 3.708189655172414,
 (2018, 'BACTERIOLOGÍA'): 3.756,
 (2018, 'BIOLOGÍA'): 3.55,
 (2018, 'ENFERMERÍA'): 3.5875000000000004,
 (2018, 'ESTADÍSTICA'): 3.6999999999999997,
 (2018, 'GEOGRAFÍA'): 3.5864864864864865,
 (2018, 'INGENIERÍA AGRONÓMICA'): 3.4768292682926827,
 (2018, 'INGENIERÍA AMBIENTAL'): 3.5393617021276595,
 (2018, 'INGENIERÍA DE ALIMENTOS'): 3.5611111111111113,
 (2018, 'INGENIERÍA DE SISTEMAS'): 3.4355263157894735,
 (2018, 'INGENIERÍA INDUSTRIAL'): 3.602777777777778,
 (2018, 'INGENIERÍA MECÁNICA'): 3.545238095238095,
 (2018,
  'LICENCIATURA EN CIENCIAS NATURALES Y EDUCACIÓN AMBIENTAL'): 3.8012048192771086,
 (2018,
  'LICENCIATURA EN EDUCACIÓN BÁSICA CON ÉNFASIS EN ARTÍSTICA'): 3.889473684210526,
 (2018,
  'LICENCIATURA EN EDUCACIÓN BÁSICA CON ÉNFASIS EN CIENCIAS SOCIALES'): 3.666,
 (2018,
  'LICENCIATURA EN EDUCACI

In [ ]:
# Ahora, utiliza el método fillna para imputar los valores perdidos en 'Promedio_Acumulado' con los promedios correspondientes
DataVA_Final['Promedio_Acumulado'].fillna(DataVA_Final.apply(lambda row: promedio_imputacion.get((row['Anosbpro'], row['Estu Prgm Academico']), np.nan), axis=1), inplace=True)

<ipython-input-23-3b204e21ad13>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  DataVA_Final['Promedio_Acumulado'].fillna(DataVA_Final.apply(lambda row: promedio_imputacion.get((row['Anosbpro'], row['Estu Prgm Academico']), np.nan), axis=1), inplace=True)


In [ ]:
DataVA_Final['Promedio_Acumulado'].isna().sum()

0

In [ ]:
DataVA_Final['Estu Estadocivil'].isna().sum()

14901

In [ ]:
DataVA_Final['Estado_Civil'].isna().sum()

894

In [ ]:
DataVA_Final['Estu Genero'].isna().sum()

0

In [ ]:
DataVA_Final['Fami Estratovivienda'].isna().sum()

135

In [ ]:
DataVA_Final['Estrato'].isna().sum()

1050

In [ ]:
DataVA_Final['Departamento'].isna().sum()

1050

In [ ]:
DataVA_Final['Ciudad'].isna().sum()

1050

In [ ]:
DataVA_Final['Estu Pagomatriculabeca'].isna().sum()

4473

In [ ]:
DataVA_Final.head()

Anosbpro Categoria_val_agreg  \
0      2018               Va-In   
1      2018               Va-Lc   
2      2018               Va-Rc   
3      2018               Va-In   
4      2018               Va-Lc   

                                 Estu Prgm Academico  valor_agreg_categoria  \
0  LICENCIATURA EN EDUCACIÓN BÁSICA CON ÉNFASIS E...             -44.351329   
1  LICENCIATURA EN EDUCACIÓN BÁSICA CON ÉNFASIS E...             -48.781156   
2  LICENCIATURA EN EDUCACIÓN BÁSICA CON ÉNFASIS E...             -36.705174   
3  ADMINISTRACIÓN EN FINANZAS Y NEGOCIOS INTERNAC...               7.242769   
4  ADMINISTRACIÓN EN FINANZAS Y NEGOCIOS INTERNAC...              -2.997487   

  Cole Naturaleza     Edadsbpro     Estu Areareside Estu Estadocivil  \
0         OFICIAL  21 o 22 Años          Area Rural          Soltero   
1         OFICIAL  21 o 22 Años          Area Rural          Soltero   
2         OFICIAL  21 o 22 Años          Area Rural          Soltero   
3         OFICIAL  21 o 22 Años  Cabecera Municipal          Soltero   
4         OFICIAL  21 o 22 Años  Cabecera Municipal          Soltero   

  Estu Genero Estu Pagomatriculabeca                 Fami Educacionmadre  \
0           M                     Si                   Primaria completa   
1           M                     Si                   Primaria completa   
2           M                     Si                   Primaria completa   
3           M                     Si  Secundaria (Bachillerato) completa   
4           M                     Si  Secundaria (Bachillerato) completa   

                  Fami Educacionpadre Fami Estratovivienda  \
0                   Primaria completa            Estrato 1   
1                   Primaria completa            Estrato 1   
2                   Primaria completa            Estrato 1   
3  Secundaria (Bachillerato) completa            Estrato 2   
4  Secundaria (Bachillerato) completa            Estrato 2   

  Fami Ingresofmiliarmensual Fami Tienecomputador Fami Tieneinternet  \
0            Menos de 1 SMLV                   No                 No   
1            Menos de 1 SMLV                   No                 No   
2            Menos de 1 SMLV                   No                 No   
3            Menos de 1 SMLV                   Si                 Si   
4            Menos de 1 SMLV                   Si                 Si   

   Indice_crecimiento          Inst Nombre Institucion  \
0            0.300000  UNIVERSIDAD DE CORDOBA-MONTERIA   
1            0.300000  UNIVERSIDAD DE CORDOBA-MONTERIA   
2            0.350000  UNIVERSIDAD DE CORDOBA-MONTERIA   
3            0.548872  UNIVERSIDAD DE CORDOBA-MONTERIA   
4            0.518797  UNIVERSIDAD DE CORDOBA-MONTERIA   

   Mod Competen Ciudada Punt  Mod Comuni Escrita Punt  Mod Ingles Punt  \
0                       99.0                     57.0             79.0   
1                       99.0                     57.0             79.0   
2                       99.0                     57.0             79.0   
3                      156.0                     65.0            149.0   
4                      156.0                     65.0            149.0   

   Mod Lectura Critica Punt  Mod Razona Cuantitat Punt  Punt C Naturales  \
0                      96.0                        104                58   
1                      96.0                        104                58   
2                      96.0                        104                58   
3                     150.0                        159                65   
4                     150.0                        159                65   

   Punt Ingles  Punt Lectura Critica  Punt Matematicas  \
0           40                    56                50   
1           40                    56                50   
2           40                    56                50   
3           44                    58                53   
4           44                    58                53   

   Punt Sociales Ciud

Ahora reemplacemos los valores faltantes NaN en la variable 'Estado_Civil' de nuestra base de datos DataVA_Final con los valores correspondientes de la variable 'Estu Estadocivil', es decir, llenaremos los valores faltantes en 'Estado_Civil' con los de 'Estu Estadocivil' solo cuando haya valores válidos en 'Estu Estadocivil' para el mismo estudiante.

In [ ]:
DataVA_Final['Estado_Civil'] = DataVA_Final['Estado_Civil'].combine_first(DataVA_Final['Estu Estadocivil'])


<ipython-input-26-4a7511af6897>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  DataVA_Final['Estado_Civil'] = DataVA_Final['Estado_Civil'].combine_first(DataVA_Final['Estu Estadocivil'])


Note que, después de haber reemplazado aquellos valores NaN en la variable 'Estado_Civil' para los que coincidian con 'Estu Estadocivil' en el cruce se reemplazaron 156 filas. Ahora para los 894 NaN restantes imputaremos estos valores faltantes en la variable 'Estado_Civil' con el valor específico "No sabe No responde(NS-NR)".

In [ ]:
DataVA_Final['Estado_Civil'].fillna("NS-NR", inplace=True)


<ipython-input-27-91a481b5b43e>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  DataVA_Final['Estado_Civil'].fillna("NS-NR", inplace=True)


In [ ]:
DataVA_Final['Estado_Civil'].isna().sum()

0

Ahora reemplacemos los valores faltantes NaN en la variable 'Fami Estratovivienda' de nuestra base de datos DataVA_Final con los valores correspondientes de la variable 'Estrato', es decir, llenaremos los valores faltantes en 'Fami Estratovivienda' con los de 'Estrato' solo cuando haya valores válidos en 'Fami Estratovivienda' para el mismo estudiante.


In [ ]:
DataVA_Final['Fami Estratovivienda'] = DataVA_Final['Fami Estratovivienda'].combine_first(DataVA_Final['Estrato'])

<ipython-input-29-14dc81c121d7>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  DataVA_Final['Fami Estratovivienda'] = DataVA_Final['Fami Estratovivienda'].combine_first(DataVA_Final['Estrato'])


Note que, después de haber reemplazado aquellos valores NaN en la variable 'Fami Estratovivienda' para los que coincidian con 'Estrato' en el cruce se reemplazaron 123 filas. Ahora para los 12 NaN restantes imputaremos estos valores faltantes en la variable 'Fami Estratovivienda' con la moda, es decir, el estrato que más se repite".


In [ ]:
mode_estrato = DataVA_Final['Fami Estratovivienda'].mode()[0]
DataVA_Final['Fami Estratovivienda'].fillna(mode_estrato, inplace=True)


<ipython-input-30-f6afc15431ca>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  DataVA_Final['Fami Estratovivienda'].fillna(mode_estrato, inplace=True)


In [ ]:
DataVA_Final['Fami Estratovivienda'].isna().sum()

0

Veamos la frecuencia de los departamentos en la variable 'Departamento'.

In [ ]:
frecuen_depart = DataVA_Final['Departamento'].value_counts()
print(frecuen_depart)


CÓRDOBA                       16050
SUCRE                           681
ANTIOQUIA                       624
BOLÍVAR                         402
ATLÁNTICO                       249
BOGOTÁ DISTRITO CAPITAL         243
CUNDINAMARCA                     75
MAGDALENA                        63
CESAR                            63
VALLE DEL CAUCA                  39
LA GUAJIRA                       36
NORTE DE SANTANDER               21
NARIÑO                           12
SANTANDER                        12
CHOCÓ                             9
TOLIMA                            9
ARCHIPIÉLAGO DE SAN ANDRÉS        6
BOYACÁ                            6
CAQUETÁ                           6
HUILA                             3
AMAZONAS                          3
ZULIA                             3
CASANARE                          3
QUINDIO                           3
CAUCA                             3
META                              3
GUAVIARE                          3
Name: Departamento, dtype: i

Ahora tomemos los 6 departamentos más importantes y el resto lo reempazaremos con "Otros departamentos".

In [ ]:
# Los 6 departamentos más importantes
top_departamentos = DataVA_Final['Departamento'].value_counts().head(6).index

# Reemplazemos los departamentos que no están en el top 6 con "Otros departamentos"
DataVA_Final['Departamento'] = DataVA_Final['Departamento'].apply(lambda x: x if x in top_departamentos else 'Otros departamentos')


<ipython-input-33-fcc267fad623>:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  DataVA_Final['Departamento'] = DataVA_Final['Departamento'].apply(lambda x: x if x in top_departamentos else 'Otros departamentos')


In [ ]:
frecuen_depart = DataVA_Final['Departamento'].value_counts()
print(frecuen_depart)

CÓRDOBA                    16050
Otros departamentos         1431
SUCRE                        681
ANTIOQUIA                    624
BOLÍVAR                      402
ATLÁNTICO                    249
BOGOTÁ DISTRITO CAPITAL      243
Name: Departamento, dtype: int64


Para los valores NaN en la variable 'Departamento' los reemplazaremos con Otros departamentos que podrian ser los falta de información o corresponder a los otros departamentos.

In [ ]:
DataVA_Final['Departamento'].fillna("Otros departamentos", inplace=True)

<ipython-input-35-43ba80fa0c83>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  DataVA_Final['Departamento'].fillna("Otros departamentos", inplace=True)


In [ ]:
DataVA_Final['Departamento'].isna().sum()

0

Veamos la frecuencia de los municipios en la variable 'Ciudad'.

In [ ]:
frecuen_munic = DataVA_Final['Ciudad'].value_counts()
print(frecuen_munic)

MONTERÍA         7089
CERETÉ           1698
LORICA           1422
SAHAGÚN           996
PLANETA RICA      816
                 ... 
PUERTO BOYACÁ       3
CAJICÁ              3
MUTATÁ              3
BELTRÁN             3
OVEJAS              3
Name: Ciudad, Length: 161, dtype: int64


Ahora tomemos los 6 minicipios más importantes y el resto lo reempazaremos con "Otros municipios".

In [ ]:
# Los 6 municipios más importantes
top_municipios = DataVA_Final['Ciudad'].value_counts().head(6).index

# Reemplazemos los municipios que no están en el top 6 con "Otros municipios"
DataVA_Final['Ciudad'] = DataVA_Final['Ciudad'].apply(lambda x: x if x in top_municipios else 'Otros municipios')

<ipython-input-38-410a47301857>:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  DataVA_Final['Ciudad'] = DataVA_Final['Ciudad'].apply(lambda x: x if x in top_municipios else 'Otros municipios')


In [ ]:
frecuen_munic = DataVA_Final['Ciudad'].value_counts()
print(frecuen_munic)

MONTERÍA            7089
Otros municipios    6948
CERETÉ              1698
LORICA              1422
SAHAGÚN              996
PLANETA RICA         816
CIÉNAGA DE ORO       711
Name: Ciudad, dtype: int64


In [ ]:
DataVA_Final['Ciudad'].isna().sum()

0

Ahora eliminemos las varables'Estu Estadocivil' y 'Estrato'. Puesto que ya se encuentran las variables correspondientes a eso.

In [ ]:
DataVA_Final.drop(['Estu Estadocivil', 'Estrato'], axis=1, inplace=True)


<ipython-input-42-7157a72e3993>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  DataVA_Final.drop(['Estu Estadocivil', 'Estrato'], axis=1, inplace=True)


In [ ]:
DataVA_Final.drop(['No SNP'], axis=1, inplace=True)

<ipython-input-43-4c9620cc7b43>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  DataVA_Final.drop(['No SNP'], axis=1, inplace=True)


In [ ]:
DataVA_Final.drop(['Categoria_val_agreg','Cole Naturaleza','Sede','Mod Competen Ciudada Punt','Mod Comuni Escrita Punt','Mod Ingles Punt',
                   'Mod Lectura Critica Punt',	'Mod Razona Cuantitat Punt',	'Punt C Naturales',	'Punt Ingles',	'Punt Lectura Critica',
                   'Punt Matematicas',	'Punt Sociales Ciudadanas',	'Puntaje,Global,SB11','Categoria','Puntaje','Ubicacion_Semestral'], axis=1, inplace=True)

<ipython-input-44-2c6ea0d81884>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  DataVA_Final.drop(['Categoria_val_agreg','Cole Naturaleza','Sede','Mod Competen Ciudada Punt','Mod Comuni Escrita Punt','Mod Ingles Punt',


In [ ]:
DataVA_Final.head()

Anosbpro                                Estu Prgm Academico  \
0      2018  LICENCIATURA EN EDUCACIÓN BÁSICA CON ÉNFASIS E...   
1      2018  LICENCIATURA EN EDUCACIÓN BÁSICA CON ÉNFASIS E...   
2      2018  LICENCIATURA EN EDUCACIÓN BÁSICA CON ÉNFASIS E...   
3      2018  ADMINISTRACIÓN EN FINANZAS Y NEGOCIOS INTERNAC...   
4      2018  ADMINISTRACIÓN EN FINANZAS Y NEGOCIOS INTERNAC...   

   valor_agreg_categoria     Edadsbpro     Estu Areareside Estu Genero  \
0             -44.351329  21 o 22 Años          Area Rural           M   
1             -48.781156  21 o 22 Años          Area Rural           M   
2             -36.705174  21 o 22 Años          Area Rural           M   
3               7.242769  21 o 22 Años  Cabecera Municipal           M   
4              -2.997487  21 o 22 Años  Cabecera Municipal           M   

  Estu Pagomatriculabeca                 Fami Educacionmadre  \
0                     Si                   Primaria completa   
1                     Si                   Primaria completa   
2                     Si                   Primaria completa   
3                     Si  Secundaria (Bachillerato) completa   
4                     Si  Secundaria (Bachillerato) completa   

                  Fami Educacionpadre Fami Estratovivienda  \
0                   Primaria completa            Estrato 1   
1                   Primaria completa            Estrato 1   
2                   Primaria completa            Estrato 1   
3  Secundaria (Bachillerato) completa            Estrato 2   
4  Secundaria (Bachillerato) completa            Estrato 2   

  Fami Ingresofmiliarmensual Fami Tienecomputador Fami Tieneinternet  \
0            Menos de 1 SMLV                   No                 No   
1            Menos de 1 SMLV                   No                 No   
2            Menos de 1 SMLV                   No                 No   
3            Menos de 1 SMLV                   Si                 Si   
4            Menos de 1 SMLV                   Si                 Si   

   Indice_crecimiento          Inst Nombre Institucion Estado_Civil  \
0            0.300000  UNIVERSIDAD DE CORDOBA-MONTERIA   SOLTERO(A)   
1            0.300000  UNIVERSIDAD DE CORDOBA-MONTERIA   SOLTERO(A)   
2            0.350000  UNIVERSIDAD DE CORDOBA-MONTERIA   SOLTERO(A)   
3            0.548872  UNIVERSIDAD DE CORDOBA-MONTERIA   SOLTERO(A)   
4            0.518797  UNIVERSIDAD DE CORDOBA-MONTERIA   SOLTERO(A)   

  Departamento            Ciudad      Caracter  Promedio_Acumulado  \
0      BOLÍVAR  Otros municipios       PUBLICA                 3.7   
1      BOLÍVAR  Otros municipios       PUBLICA                 3.7   
2      BOLÍVAR  Otros municipios       PUBLICA                 3.7   
3      CÓRDOBA      PLANETA RICA  NO REPORTADO                 3.5   
4      CÓRDOBA      PLANETA RICA  NO REPORTADO                 3.5   

                  Modalidad Acceso_Internet Portatil_PC Celular_Tablet  \
0  UNIVERSITARIO PRESENCIAL      NO REPORTA  NO REPORTA     NO REPORTA   
1  UNIVERSITARIO PRESENCIAL      NO REPORTA  NO REPORTA     NO REPORTA   
2  UNIVERSITARIO PRESENCIAL      NO REPORTA  NO REPORTA     NO REPORTA   
3   UNIVERSITARIO DISTANCIA      NO REPORTA  NO REPORTA     NO REPORTA   
4   UNIVERSITARIO DISTANCIA      NO REPORTA  NO REPORTA     NO REPORTA   

  Grupo_Etnico Persona_Con_Discapacidad Capacidad_Excepcional  \
0   NO REPORTA               NO REPORTA            NO REPORTA   
1   NO REPORTA               NO REPORTA            NO REPORTA   
2   NO REPORTA               NO REPORTA            NO REPORTA   
3   NO REPORTA               NO REPORTA            NO REPORTA   
4   NO REPORTA               NO REPORTA            NO REPORTA   

  Poblacion_Vulnerable Inmigrantes Tipo_Discapacidad Tiene_Hijos  \
0           NO REPORTA  NO REPORTA        NO REPORTA  NO REPORTA   
1           NO REPORTA  NO REPORTA        NO REPORTA  NO REPORTA   
2           NO REPORTA  NO REPORTA        NO REPORTA  NO REPORTA   
3           NO

In [ ]:
frecuen_Beca = DataVA_Final['Estu Pagomatriculabeca'].value_counts()
print(frecuen_Beca)

No    11436
Si     3771
Name: Estu Pagomatriculabeca, dtype: int64


In [ ]:
mode_Beca = DataVA_Final['Estu Pagomatriculabeca'].mode()[0]
DataVA_Final['Estu Pagomatriculabeca'].fillna(mode_Beca, inplace=True)


<ipython-input-47-9b817e0d1b22>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  DataVA_Final['Estu Pagomatriculabeca'].fillna(mode_Beca, inplace=True)


In [ ]:
frecuen_Beca = DataVA_Final['Estu Pagomatriculabeca'].value_counts()
print(frecuen_Beca)

No    15909
Si     3771
Name: Estu Pagomatriculabeca, dtype: int64


In [ ]:
DataVA_Final['Estu Pagomatriculabeca'].isna().sum()

0

In [ ]:
# Porcentajes de valores nulos por columna
null_percentage = (DataVA_Final.isnull().sum() / len(DataVA_Final)) * 100

null_percentage_df = pd.DataFrame({'Columna': null_percentage.index, 'Porcentaje de Nulos': null_percentage.values})

# Ordenamos el DataFrame por el porcentaje de nulos de mayor a menor
null_percentage_df = null_percentage_df.sort_values(by='Porcentaje de Nulos', ascending=False)

print(null_percentage_df)

                            Columna  Porcentaje de Nulos
12               Fami Tieneinternet            22.728659
3                         Edadsbpro            22.728659
7               Fami Educacionmadre            22.728659
8               Fami Educacionpadre            22.728659
10       Fami Ingresofmiliarmensual            22.728659
11             Fami Tienecomputador            22.728659
31  Consume_Sustancias_Psicoactivas             5.335366
21                  Acceso_Internet             5.335366
22                      Portatil_PC             5.335366
23                   Celular_Tablet             5.335366
24                     Grupo_Etnico             5.335366
25         Persona_Con_Discapacidad             5.335366
26            Capacidad_Excepcional             5.335366
20                        Modalidad             5.335366
27             Poblacion_Vulnerable             5.335366
28                      Inmigrantes             5.335366
29                Tipo_Discapac

Variables Dummy para todas las columnas categóricas.

In [ ]:
DataVA_Final.drop(['Fami Tienecomputador', 'Fami Tieneinternet'], axis=1, inplace=True)

<ipython-input-53-a7fae788801b>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  DataVA_Final.drop(['Fami Tienecomputador', 'Fami Tieneinternet'], axis=1, inplace=True)


In [ ]:
DataVA_Final_Final = pd.get_dummies(DataVA_Final, columns=['Estu Prgm Academico', 'Estu Areareside', 'Estu Genero',
                                                     'Estu Pagomatriculabeca', 'Fami Educacionmadre','Fami Educacionpadre','Fami Ingresofmiliarmensual',
                                                     'Estado_Civil', 'Departamento', 'Ciudad', 'Caracter','Modalidad', 'Acceso_Internet', 'Portatil_PC',
                                                     'Celular_Tablet', 'Grupo_Etnico', 'Persona_Con_Discapacidad','Capacidad_Excepcional',
                                                     'Poblacion_Vulnerable', 'Inmigrantes', 'Tipo_Discapacidad', 'Tiene_Hijos',
                                                     'Consume_Sustancias_Psicoactivas'], drop_first=False)


In [ ]:
DataVA_Final_Final.head()

Anosbpro  valor_agreg_categoria     Edadsbpro Fami Estratovivienda  \
0      2018             -44.351329  21 o 22 Años            Estrato 1   
1      2018             -48.781156  21 o 22 Años            Estrato 1   
2      2018             -36.705174  21 o 22 Años            Estrato 1   
3      2018               7.242769  21 o 22 Años            Estrato 2   
4      2018              -2.997487  21 o 22 Años            Estrato 2   

   Indice_crecimiento          Inst Nombre Institucion  Promedio_Acumulado  \
0            0.300000  UNIVERSIDAD DE CORDOBA-MONTERIA                 3.7   
1            0.300000  UNIVERSIDAD DE CORDOBA-MONTERIA                 3.7   
2            0.350000  UNIVERSIDAD DE CORDOBA-MONTERIA                 3.7   
3            0.548872  UNIVERSIDAD DE CORDOBA-MONTERIA                 3.5   
4            0.518797  UNIVERSIDAD DE CORDOBA-MONTERIA                 3.5   

   Estu Prgm Academico_ACUICULTURA  \
0                                0   
1                                0   
2                                0   
3                                0   
4                                0   

   Estu Prgm Academico_ADMINISTRACIÓN EN FINANZAS Y NEGOCIOS INTERNACIONALES  \
0                                                  0                           
1                                                  0                           
2                                                  0                           
3                                                  1                           
4                                                  1                           

   Estu Prgm Academico_ADMINISTRACIÓN EN SALUD  \
0                                            0   
1                                            0   
2                                            0   
3                                            0   
4                                            0   

   Estu Prgm Academico_BACTERIOLOGÍA  Estu Prgm Academico_BIOLOGÍA  \
0                                  0                             0   
1                                  0                             0   
2                                  0                             0   
3                                  0                             0   
4                                  0                             0   

   Estu Prgm Academico_DERECHO  Estu Prgm Academico_ENFERMERÍA  \
0                            0                               0   
1                            0                               0   
2                            0                               0   
3                            0                               0   
4                            0                               0   

   Estu Prgm Academico_ESTADÍSTICA  Estu Prgm Academico_FÍSICA  \
0                                0                           0   
1                                0                           0   
2                                0                           0   
3                                0                           0   
4                                0                           0   

   Estu Prgm Academico_GEOGRAFÍA  Estu Prgm Academico_INGENIERÍA AGRONÓMICA  \
0                              0                                          0   
1                              0                                          0   
2                              0                                          0   
3                              0                                          0   
4                              0                                          0   

   Estu Prgm Academico_INGENIERÍA AMBIENTAL  \
0                                         0   
1                                         0   
2                                         0   
3                                         0   
4                                         0   

   Estu Prgm Academico_INGENIERÍA DE ALIMENTOS  \
0                                            0   
1          

In [ ]:
frecuen_B = DataVA_Final['Fami Ingresofmiliarmensual'].value_counts()
print(frecuen_B)

Menos de 1 SMLV               6498
Entre 1 y menos de 2 SMLV     5775
Entre 2 y menos de 3 SMLV     1947
Entre 3 y menos de 5 SMLV      780
Entre 5 y menos de 7 SMLV      144
Entre 7 y menos de 10 SMLV      45
10 o más SMLV                   18
Name: Fami Ingresofmiliarmensual, dtype: int64


In [ ]:
# Variables independientes (características)
X = DataVA.drop(['valor_agreg_categoria'], axis=1)  # Excluye la columna de la variable objetivo

# Variable objetivo (Y)
y = DataVA['valor_agreg_categoria']  # Reemplaza 'valor_agreg_categoria' con el nombre de tu variable objetivo

In [ ]:
X.head()

Anosbpro Categoria_val_agreg  \
0      2018               Va-In   
1      2018               Va-Lc   
2      2018               Va-Rc   
3      2018               Va-In   
4      2018               Va-Lc   

                                 Estu Prgm Academico  Anosb11  Añosb112  \
0  LICENCIATURA EN EDUCACIÓN BÁSICA CON ÉNFASIS E...     2013     20132   
1  LICENCIATURA EN EDUCACIÓN BÁSICA CON ÉNFASIS E...     2013     20132   
2  LICENCIATURA EN EDUCACIÓN BÁSICA CON ÉNFASIS E...     2013     20132   
3  ADMINISTRACIÓN EN FINANZAS Y NEGOCIOS INTERNAC...     2013     20132   
4  ADMINISTRACIÓN EN FINANZAS Y NEGOCIOS INTERNAC...     2013     20132   

  Cole Caracter Cole Naturaleza     Edadsbpro        Epcc        Epce  \
0     ACADÉMICO         OFICIAL  21 o 22 Años  133.124611  144.908047   
1     ACADÉMICO         OFICIAL  21 o 22 Años  133.124611  144.908047   
2     ACADÉMICO         OFICIAL  21 o 22 Años  133.124611  144.908047   
3     ACADÉMICO         OFICIAL  21 o 22 Años  150.570796  148.259827   
4     ACADÉMICO         OFICIAL  21 o 22 Años  150.570796  148.259827   

         Epin        Eplc        Eprc     Estu Areareside Estu Consecutivo  \
0  123.351329  144.781156  140.705174          Area Rural   EK201830168341   
1  123.351329  144.781156  140.705174          Area Rural   EK201830168341   
2  123.351329  144.781156  140.705174          Area Rural   EK201830168341   
3  141.757231  152.997487  157.645090  Cabecera Municipal   EK201830178478   
4  141.757231  152.997487  157.645090  Cabecera Municipal   EK201830178478   

  Estu Consecutivo 11 Estu Depto Reside Estu Estadocivil Estu Fechanacimiento  \
0    SB11201320062403           CORDOBA          Soltero           1997-03-09   
1    SB11201320062403           CORDOBA          Soltero           1997-03-09   
2    SB11201320062403           CORDOBA          Soltero           1997-03-09   
3    SB11201320216423           CORDOBA          Soltero           1997-10-31   
4    SB11201320216423           CORDOBA          Soltero           1997-10-31   

  Estu Genero Estu Horassemanatrabaja Estu Mcpio Reside  Estu Nse Ies  \
0           M         Más de 30 horas        SAN PELAYO             1   
1           M         Más de 30 horas        SAN PELAYO             1   
2           M         Más de 30 horas        SAN PELAYO             1   
3           M     Entre 11 y 20 horas      PLANETA RICA             1   
4           M     Entre 11 y 20 horas      PLANETA RICA             1   

   Estu Nse Individual Estu Nucleo Pregrado Estu Pagomatriculabeca  \
0                  1.0            EDUCACIÓN                     Si   
1                  1.0            EDUCACIÓN                     Si   
2                  1.0            EDUCACIÓN                     Si   
3                  2.0       ADMINISTRACIÓN                     Si   
4                  2.0       ADMINISTRACIÓN                     Si   

  Estu Pagomatriculacredito Estu Pagomatriculapadres Estu Pagomatriculapropio  \
0                        No                       Si                       No   
1                        No                       Si                       No   
2                        No                       Si                       No   
3                        Si                       Si                       No   
4                        Si                       Si                       No   

   Fami Cuartoshogar                 Fami Educacionmadre  \
0                4.0                   Primaria completa   
1                4.0                   Primaria completa   
2                4.0                   Primaria completa   
3                3.0  Secundaria (Bachillerato) completa   
4                3.0  Secundaria (Bachillerato) completa   

                  Fami Educacionpadre Fami Estratovivienda  \
0                   Primaria completa            Estrato 1   
1                   Primaria completa            Estrato 1   
2                   Primaria completa            Est

In [ ]:
y.head()

0   -44.351329
1   -48.781156
2   -36.705174
3     7.242769
4    -2.997487
Name: valor_agreg_categoria, dtype: float64